implementation/src/main/java/io/smallrye/config/ConfigValueProperties.java

```diff
@@ -0,0 +1,283 @@
+package io.smallrye.config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.util.HashMap;
+
+class ConfigValueProperties extends HashMap<String, ConfigValue> {
+    public synchronized void load(Reader reader) throws IOException {
+        load0(new LineReader(reader));
+    }
+
+    public synchronized void load(InputStream inStream) throws IOException {
+        load0(new LineReader(inStream));
+    }
+
+    private void load0(LineReader lr) throws IOException {
```

Comment:
This would take a fairly long time to analyze in depth; it sufficies to have test cases for each state though.